modules/external_petsc_solver/include/problem/ExternalPETScProblem.h

```diff
@@ -46,7 +46,6 @@ class ExternalPETScProblem : public ExternalProblem
   /// PETSc solver
   TS & _ts;
   /// PETSc solver solution
-  Vec _petsc_sol;
+  Vec & _petsc_sol;
```

Comment:
Why this change? We need a reference to a pointer?

Reply:
I want app to hold solution. Right, we could simply use a pointer. We could simply consider `Vec` as a class, and more readable for me.

Reply:
So you want the `App` to hold the solution and not the `Problem`? And you're arguing that I shouldn't make use of my  knowledge that `Vec` is actually a pointer? That's a fair argument. Ok I support this change.